profile/pseudo_locs_test.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package profile
+
+import (
+	"reflect"
+	"testing"
+)
+
+func countSampleLocs(p *Profile) []int {
+	var locs []int
+	for _, s := range p.Sample {
+		locs = append(locs, len(s.Location))
+	}
+	return locs
+}
+
+func TestAddPseudoLocs(t *testing.T) {
+	// Perform several forms of pseudo location addition on the test
+	// profile.
+	notfound := "notfound"
+	key1 := "key1"
+	key2 := "key2"
+	type pseudoTestcase struct {
+		name             string
+		rootTag, leafTag string
+		wr, wl           bool
+		locs             []int
+	}
+	for _, tc := range []pseudoTestcase{
+		// Empty root and leaf tags.
+		{"EmptyRootAndLeafTags", "", "", false, false, []int{1, 2, 2, 2, 2}},
+		// Root and leaf tags not present in any of the samples.
+		{"TagsNotFound", notfound, notfound, false, false, []int{1, 2, 2, 2, 2}},
+		// One root tag.
+		{"AddRootTag", key1, "", true, false, []int{2, 3, 3, 3, 3}},
+		// One leaf tag.
+		{"AddLeafTag", "", key2, false, true, []int{2, 2, 3, 3, 3}},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			prof := testProfile1.Copy()
+			gr, gl := prof.AddPseudoLocs(tc.rootTag, tc.leafTag)
+			if gr != tc.wr || gl != tc.wl {
+				t.Fatalf("Got AddPseudoLocs(%s, %s)=%v,%v want %v,%v", tc.rootTag, tc.leafTag, gr, gl, tc.wr, tc.wl)
```

Comment:
nit: please follow https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures recommendations for the error string. E.g. function name with inputs typically goes first, without the "Got " prefix.

Reply:
Done